### PR TITLE
 Add ReflectionTypeLoadException tests

### PR DESCRIPTION
**Pull request by wtgodbe**
Regression tests for https://github.com/dotnet/coreclr/issues/15710

@jkotas PTAL

### CODE REVIEW

**Comment by wtgodbe** [comment]
Linux test failure also appears in https://ci3.dot.net/job/dotnet_corefx/job/master/job/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/6953/, which is for a different PR, so appears unrelated. Testing again anyways.

**Comment by wtgodbe** [comment]
@dotnet-bot test this please

File: src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        }
+
+        [Fact]
+        public static void NullArgumentsNoNullPointerException()
+        {
+            bool foundRtleException = false;
+            try
+            {
+                throw new ReflectionTypeLoadException(null, null, "Null arguments");
+
```
Comment (jkotas) [inline src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs]: Extra line

File: src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            }
+            catch (ReflectionTypeLoadException)
+            {
+                foundRtleException = true;
```
Comment (jkotas) [inline src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs]: This should call ToString and Message to exercise the codepaths that you have modified (and that you had a bug in originally).

File: src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+{
+    public static class ReflectionTypeLoadExceptionTests
+    {
+        [Fact]
+        public static void NullExceptionsNoNullPointerException()
+        {
+            bool foundRtleException = false;
+            try
+            {
+                Type[] Typo = new Type[1];
```
Comment (jkotas) [inline src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs]: The coding convention is for locals to start with lowercase.

**Comment by wtgodbe** [comment]
@jkotas updated to address feedback, does it look better?

File: src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+namespace System.Reflection.Tests
+{
+    public static class ReflectionTypeLoadExceptionTests
+    {
+        [Fact]
+        public static void NullExceptionsNoNullPointerException()
+        {
+            try
+            {
+                Type[] typo = new Type[1];
```
Comment (jkotas) [inline src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs]: Do we have a prior art in corefx for using Spanish(?) names for locals?

File: src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                Exception[] excepto = new Exception[1];
+                throw new ReflectionTypeLoadException(typo, excepto, "Null elements in Exceptions array");
+            }
+            catch (ReflectionTypeLoadException e)
+            {
+                Assert.NotNull(e.ToString());
```
Comment (jkotas) [inline src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs]: You can also assert here that the LoaderExceptions and Types properties are what you expect them to be.

File: src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        [Fact]
+        public static void NullArgumentsNoNullPointerException()
+        {
+            try
+            {
+                throw new ReflectionTypeLoadException(null, null, "Null arguments");
+            }
+            catch (ReflectionTypeLoadException e)
+            {
+                Assert.NotNull(e.ToString());
```
Comment (jkotas) [inline src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs]: Same here.

File: src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        [Fact]
+        public static void NullExceptionsNoNullPointerException()
+        {
+            try
+            {
+                Type[] typo = new Type[1];
+                Exception[] excepto = new Exception[1];
+                throw new ReflectionTypeLoadException(typo, excepto, "Null elements in Exceptions array");
+            }
+            catch (ReflectionTypeLoadException e)
```
Comment (jkotas) [inline src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs]: You do need to actually throw and catch the exception for this validation. You can just create the exception without throwing it. E.g. it is what AggregateException tests do: https://github.com/dotnet/corefx/blob/master/src/System.Threading.Tasks/tests/AggregateExceptionTests.cs#L17

**Comment by wtgodbe** [comment]
Updated again.

**Comment by wtgodbe** [comment]
@dotnet-bot test Linux x64 Release Build

File: src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        [Fact]
+        public static void NullExceptionsNoNullPointerException()
+        {
+            Type[] types = new Type[1];
+            Exception[] exceptions = new Exception[1];
+            ReflectionTypeLoadException rtle = new ReflectionTypeLoadException(types, exceptions, "Null elements in Exceptions array");
+            Assert.NotNull(rtle.ToString());
+            Assert.NotNull(rtle.Message);
+            Assert.Equal(rtle.LoaderExceptions.Length, 1);
```
Comment (justinvp) [inline src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs]: The expected value should be the first parameter:

```c#
Assert.Equal(1, rtle.LoaderExceptions.Length);
```

Applies throughout this PR.

Comment (justinvp) [reply src/System.Runtime/tests/System/Reflection/ReflectionTypeLoadExceptionTests.cs]: Opened #26166 to address.

**Pull request merged**